applications/nrf_desktop/doc/leds.rst

```diff
@@ -0,0 +1,43 @@
+.. leds:
+
+leds module
+###########
+
+``leds`` module is responsible for controlling LED diodes (setting LED effect in reaction to ``led_event``).
+
+
+Configuration
+*************
+
+For setting brightness ``leds`` uses PWM driver.
+In order to use the module, following configuration options have to be set:
+
+- :option:`CONFIG_PWM`
+- :option:`CONFIG_PWM_0`
+  0 denotes PWM port number. All PWM ports used by application should be configured.
+
+For current implementation :option:`CONFIG_PWM_NRFX` is set and :option:`CONFIG_PWM_NRF5_SW` is unset (in order to use hardware PWM driver).
+
+Module is enabled and configured using Kconfig options (details described as help).
+
+File ``leds_def.h`` is used to define mapping PWM channels to pin numbers.
+Current implementation uses symbols defined in device tree.
+These symbols are defined for board, but application may overlay definining own ``dts.overlay`` file in the application configuration directory dedicated for this board.
+
+
+Implementation details
+**********************
+
+Module sets effect for given LED based on ``led_event``.
+In addition, it turns off all LEDs when going to power down state.
+In such case, PWM drivers are also set to suspended state (in order to reduce power consumption).
+While application goes to error state, LEDs are still used to indicate error.
+
+To achieve selected LED effect, colors for given LED are periodically updated using work (:c:type:`k_delayed_work`).
```

Comment:
```suggestion
To achieve the selected LED effect, colors for the given LED are periodically updated using work (:c:type:`k_delayed_work`).
```
What is this `work`? Also, you mention `selected LED effect` -- where do I select it?

Reply:
selected with recent `led_event` for given LED. Work is reference to Zephyr work (k_delayed_work for leds module)